basic_syntax.md

```diff
@@ -65,7 +65,17 @@ html! {
 }
 ```
 
-Use `(foo)` syntax to splice in the value of `foo` at runtime. Any HTML special characters are escaped by default.
+Use `(foo)` syntax to splice in the value of `foo` at runtime. Any HTML special characters are escaped by default. Arbitrary Rust code can be included in a compound expression when a single expression would be hard to read or when Rust's type inference needs a little bit of help:
```

Comment:
* The explanation should come after the example code, not before. (I'm open to changing this, but for now we should stay consistent with the rest of the text.)

* This should also be split into multiple sentences. There are a lot of ideas packed into this sentence -- any expression can go here, blocks are expressions too, ascribe types using a let statement -- which should be elaborated upon separately.

* In Rust, the syntax construct is called a [block](https://doc.rust-lang.org/reference.html#block-expressions), not a compound expression.